script/prerender/sitemap.ts

```diff
@@ -7,25 +7,54 @@ import max from 'lodash/fp/max';
 import sitemap, { SitemapItemOptions } from 'sitemap';
 import { writeAssetFile } from './fileUtils';
 
-const sitemapPath = (pathName: string) => `/rex/sitemaps/${pathName}.xml`;
-const sitemaps: SitemapItemOptions[] = [];
+export const sitemapPath = (pathName: string) => `/rex/sitemaps/${pathName}.xml`;
 
-export const renderSitemap = (filename: string, urls: SitemapItemOptions[]) => {
+export const renderAndSaveSitemap = async(
+  saveFile: (path: string, contents: string) => void,
+  slug: string,
+  urls: SitemapItemOptions[]
+) => {
   const bookSitemap = sitemap.createSitemap({ hostname: 'https://openstax.org', urls });
+
+  const filePath = sitemapPath(slug);
+
+  await saveFile(filePath, bookSitemap.toString());
```

Comment:
do you want to type this as returning a promise?

Reply:
writeAssetFile() is not async but I suppose I can wrap it